os/tools/compression: Add gitignore for compression folder

sijupunnnoose opened the pull request:
Compression tool generates temp files and executable for
binary compression which need not to be added to git.

seinfra commented:
Target : [214d734227f5ec386e7eb86bb176be08ebaebd35] - Code Rule Check (C++) OK.

seinfra commented:
Target : [214d734227f5ec386e7eb86bb176be08ebaebd35] - Code Rule Check OK.

seinfra commented:
Target : [31af83f80fcaf28fe9fbd2509fdc87060a0404e3] - Code Rule Check (C++) OK.

seinfra commented:
Target : [31af83f80fcaf28fe9fbd2509fdc87060a0404e3] -

sunghan-chang commented:
Could you check all of copied or linked or generated files are deleted with clean command?

sijupunnnoose commented:
> Could you check all of copied or linked or generated files are deleted with clean command?

.config was missing in clean, add it in clean

seinfra commented:
Target : [703f0db21427085887bd760aebe6557d6d435ce9] - Code Rule Check (C++) OK.

seinfra commented:
Target : [703f0db21427085887bd760aebe6557d6d435ce9] - Code Rule Check OK.

sunghan-chang commented:
@sijupunnnoose
1. `distclean` target is missing either.
Because of that, it shows below:
```
make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
make[1]: *** No rule to make target 'distclean'.  Stop.
```
2. When I call clean, only tools/compression shows unnecessary information
```
make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
CP: .config
=== cleaning ===
make[1]: Leaving directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
```
Please remove them.
3. Why does it call `CP` with clean command?

sunghan-chang commented:
4. could you split commits for clean and distclean from gitignore commit?

seinfra commented:
Target : [b50d72cb3e01f863d8b8b690229603410d404103] - Code Rule Check (C++) OK.

seinfra commented:
Target : [b50d72cb3e01f863d8b8b690229603410d404103] - Code Rule Check OK.

seinfra commented:
Target : [1392ad0e5a302109fa3168b16555a9657132ca4e] - Code Rule Check OK.

seinfra commented:
Target : [1392ad0e5a302109fa3168b16555a9657132ca4e] - Code Rule Check (C++) OK.

sijupunnnoose commented:
> @sijupunnnoose
> 
> 1. `distclean` target is missing either.
>    Because of that, it shows below:
> 
> ```
> make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> make[1]: *** No rule to make target 'distclean'.  Stop.
> ```
> 
> 1. When I call clean, only tools/compression shows unnecessary information
> 
> ```
> make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> CP: .config
> === cleaning ===
> make[1]: Leaving directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> ```
> 
> Please remove them.
> 3. Why does it call `CP` with clean command?

Added distclean section and removed unnecessary information

sijupunnnoose commented:
> 1. could you split commits for clean and distclean from gitignore commit?

Done

sunghan-chang commented on os/tools/compression/Makefile:
```diff
@@ -90,29 +90,25 @@ $(APPNAME): Makefile $(CONFIG) $(CCONFIG) $(COMPHEADER) $(OBJECTS)
 	@$(CC) $(LDFLAGS) $(OBJECTS) $(LIBFILES) -o $@
```
Is this CC? not LD?

sijupunnnoose replied:
It is CC , we are building the tool for compressing image for ubuntu machine. Not cross compiling

sunghan-chang commented on os/tools/compression/Makefile:
```diff
@@ -90,29 +90,25 @@ $(APPNAME): Makefile $(CONFIG) $(CCONFIG) $(COMPHEADER) $(OBJECTS)
 # =================================
 $(CONFIG): $(TINYARADIR)/.config
-	@echo "CP: .config"
 	@cp $(TINYARADIR)/.config .
```
As I left before, when we call `clean`, this is executed. Could you check the reason of calling this with clean?

sijupunnnoose replied:
It is not calling copy now with clean.

sunghan-chang commented on os/tools/compression/Makefile:
```diff
@@ -122,3 +118,9 @@ endif
+
+distclean: clean
+	$(call DELFILE, Make.dep)
+	$(call DELFILE, .depend)
```
To use definition like DELFILE, you must include `Config.mk` by including Make.defs of os folder. But here is no including. Is this really working?

sijupunnnoose replied:
I missed to -include $(TOPDIR)/Make.defs, will correct it

sunghan-chang commented on os/tools/compression/Makefile:
```diff
@@ -122,3 +118,9 @@ endif
 	@rm -f config.h
 	@rm -f compression.h
 	@rm -f *.h
+	@rm -f .config
```
Let's use DEFFILE and DELDIR

sijupunnnoose replied:
Will change to DELFILE &DELDIR

seinfra commented:
Target : [436364ff773edafa8fb00a9ae4689e3c570b6d2d] - Code Rule Check (C++) OK.

seinfra commented:
Target : [436364ff773edafa8fb00a9ae4689e3c570b6d2d] - Code Rule Check OK.
